### PR TITLE
Users backend api

### PR DESCRIPTION
**Pull request by epugh**

### CODE REVIEW

File: app/controllers/ApiController.scala [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
-import java.time.format.DateTimeFormatter
-import java.time.LocalDateTime
-
-import scala.concurrent.{ExecutionContext, Future}
 import controllers.auth.AuthActionFactory
 import models.FeatureToggleModel.FeatureToggleService
 import models._
 import models.config.SmuiVersion
-import models.input.{InputTagId, InputValidator, ListItem, SearchInputId, SearchInputWithRules}
+import models.input._
```
Comment (epugh) [inline app/controllers/ApiController.scala:9]: Can you check your IDE?   We typically have been putting ALL the imports in, isntead of using the short cut...   This make a lot of changes on this, which makes it harder to review.

File: app/controllers/ApiController.scala [hunk 2]
```diff
@@ -90,6 +89,41 @@ class ApiController @Inject()(authActionFactory: AuthActionFactory,
...
 
+  def listAllUsers(): Action[AnyContent] = authActionFactory.getAuthenticatedAction(Action) {
+    Ok(Json.toJson(searchManagementRepository.listAllUsers()))
+  }
+
+  def getUser(userId: String): Action[AnyContent] = authActionFactory.getAuthenticatedAction(Action) {
+    Ok(Json.toJson(searchManagementRepository.getUserById(userId)))
+  }
+
+  def lookup(by: String, value: String): Action[AnyContent] = authActionFactory.getAuthenticatedAction(Action) {
```
Comment (epugh) [inline app/controllers/ApiController.scala:100]: could this be getUser() with a different params?   Or, call it lookupUser???

File: app/models/SearchManagementRepository.scala [hunk 3]
```diff
@@ -56,6 +56,26 @@ class SearchManagementRepository @Inject()(dbapi: DBApi, toggleService: FeatureT
     InputTag.insert(inputTag)
   }
 
+  def listAllUsers(): Seq[User] = db.withConnection { implicit connection =>
```
Comment (epugh) [inline app/models/SearchManagementRepository.scala:59]: I wonder about adding users and teams etc to SearchManagementRepository?  Is there another pattern we could use?   Seems likewe add EVERYTHING to APIController, then add EVERYTHING to SearchManagementRepository.

Maybe a UserManagementRepository?  Dunno, maybe worth thinking about.

File: app/models/User.scala [hunk 4]
```diff
@@ -1,15 +1,116 @@
...
+import java.sql.Connection
+import java.time.LocalDateTime
 import scala.collection.mutable
 
-case class User(name: String, email: String, password: String)
+class UserId(id: String) extends Id(id)
+object UserId extends IdObject[UserId](new UserId(_))
+
+/**
+  * Defines a tag that can be assigned to a search input
```
Comment (epugh) [inline app/models/User.scala:16]: plase update comments

File: test/models/UserSpec.scala [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import utils.WithInMemoryDB
+
+import java.time.LocalDateTime
+
+class UserSpec extends FlatSpec with Matchers with BeforeAndAfterEach with WithInMemoryDB {
+
+  private val users = Seq(
+    User.create(None, None, "good rule", exported = false),
+    User.create(None, Some("tenant"), "MO", exported = true),
+    User.create(Some(index.id), Some("tenant"), "MO_DE", exported = true),
```
Comment (epugh) [inline test/models/UserSpec.scala:15]: not sure I understand how this is working???    maybe not completed yet???

File: app/controllers/ApiController.scala (hunk 1 above)
Comment (Paul-Blanchaert) [reply app/controllers/ApiController.scala]: Reverted this import changes

File: app/controllers/ApiController.scala (hunk 2 above)
Comment (Paul-Blanchaert) [reply app/controllers/ApiController.scala]: 2 methods added for lookup of user: via username and email

File: app/models/SearchManagementRepository.scala (hunk 3 above)
Comment (Paul-Blanchaert) [reply app/models/SearchManagementRepository.scala]: Good point, I leave that decision with the technical architect...

File: app/models/User.scala (hunk 4 above)
Comment (Paul-Blanchaert) [reply app/models/User.scala]: Done

File: test/models/UserSpec.scala (hunk 5 above)
Comment (Paul-Blanchaert) [reply test/models/UserSpec.scala]: should be more reasonable now...

**Comment by epugh** [comment]
We are closing this one in favour of continuing on #85

**Pull request closed without merging**